applications/FluidDynamicsApplication/custom_elements/compressible_navier_stokes_explicit.cpp

```diff
@@ -299,9 +299,9 @@ void CompressibleNavierStokesExplicit<TDim, TNumNodes>::CalculateOnIntegrationPo
             rOutput[i_gauss] = rho_grad;
         }
     } else if (rVariable == TEMPERATURE_GRADIENT) {
-        const auto& rho_grad = CalculateMidPointTemperatureGradient();
+        const auto& temp_grad = CalculateMidPointTemperatureGradient();
```

Comment:
```suggestion
        const array_1d<double,3>& temp_grad = CalculateMidPointTemperatureGradient();
```

Reply:
Done.

Reply:
Just out of curiosity, why is it better to use `array_1d<double,3>` here?

Reply:
We experienced some weird issues in the past when autodeducing the `array_1d<double,3>` types. If I don't remember wrong, in cases like this one, there it may deduce it as a function type with array return.

Reply:
:S scary, thanks for the explanation

Reply:
(problems raise when you want to use the array in an expression like `2*my_array`)

Reply:
@ddiezrod see #7414